bot_handler.py

```diff
@@ -74,42 +82,93 @@ def on_bot_update(self, update):
         elif update.message.text == NUMBER_BUTTON:
             self._next_state = NumberState.name
 
+
+current_item = ''
+current_quantity = ''
```

Comment:
This variables shouldn't be global. It is because different users sending messages when being at the same state will result in overwriting user1 data in these variables by user2's data. In this case user1's order will be the same as user2's order and user1's original order will be lost.

As I understood from code below `current_item` and `current_quantity` needed only for sending confirmation message to the current user in RequestState2. But these variables are in fact redundant because of `user_data` variable in RequestOrder2 which already contains values of `current_item` and `current_quantity`.

Please see suggestions.